plugins/Login/Controller.php

```diff
@@ -269,27 +271,51 @@ public function confirmPassword()
      */
     public function logme()
     {
-        if (Config::getInstance()->General['login_allow_logme'] == 0) {
+        if (Config\GeneralConfig::getConfigValue('login_allow_logme') == 0) {
             throw new Exception('This functionality has been disabled in config');
         }
 
-        $password = Common::getRequestVar('password', null, 'string');
+        $request  = Request::fromRequest();
+        $password = $request->getStringParameter('password');
+        $login    = $request->getStringParameter('login');
 
-        $login = Common::getRequestVar('login', null, 'string');
-        if (Piwik::hasTheUserSuperUserAccess($login)) {
-            throw new Exception(
-                Piwik::translate('Login_ExceptionInvalidSuperUserAccessAuthenticationMethod', ["logme"])
-            );
-        }
+        $login = Access::doAsSuperUser(function () use ($login) {
+            try {
+                $user = \Piwik\Plugins\UsersManager\API::getInstance()->getUser($login);
+            } catch (\Exception $e) {
+                // if a user can't be found for any reason we throw a generic exception below to avoid enumeration
+            }
+
+            if (empty($user)) {
+                throw new Exception(Piwik::translate('Login_LoginPasswordNotCorrect'));
+            }
+
+            // Note: Not using Piwik::hasTheUserSuperUserAccess here on purpose as that would require
+            // a logged in user to work and wouldn't work correctly within Access::doAsSuperUser
```

Comment:
Not 100% clear on the comment here, it says it wouldn't work within Access::doAsSuperUser but we are in a callback of that method, can we maybe clarify this?